src/redisearch_rs/inverted_index/src/index_result.rs

```diff
@@ -41,6 +41,13 @@ unsafe extern "C" {
     /// # Safety
     /// The caller must ensure that the `t` pointer is valid and points to an `RSQueryTerm`.
     unsafe fn Term_Free(t: *mut RSQueryTerm);
+
+    /// Make a complete clone of the metrics array and increment the reference count of each value
+    ///
+    /// # Safety
+    /// The caller must ensure that the `src` pointer is valid and points to an `RSYieldableMetric`.
+    /// The caller must also not free the returned pointer.
```

Comment:
Who will free it then? :)

Reply:
Expanded in https://github.com/RediSearch/RediSearch/pull/6554/commits/23d180a3f1deae9d3845977ebf6b866ae1fcd8b0